src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -550,6 +552,11 @@ public void upgrade_noop(ChocolateyConfiguration config)
             return packageUpgrades;
         }
 
+        private void before_package_upgrade(PackageResult packageResult, ChocolateyConfiguration config)
+        {
+            _powershellService.run_package_scripts_for_command(config, packageResult, CommandNameType.upgrade);
+        }
+

```

Comment:
This action is usually run before uninstall as well, I'm imagining you haven't hit that point yet.